From 43d4109ced1e9f5d082d27ef4abe1d581d305465 Mon Sep 17 00:00:00 2001
From: Dhineshkumar M <mdhineshkumar@ami.com>
Date: Sat, 3 Feb 2024 20:07:57 +0530
Subject: [PATCH] Fix for nfs bad dbus request error

dbus varaint data type is not properly type casting for negative integer
value

Signed-off-by: Dhineshkumar M <mdhineshkumar@ami.com>
---
 redfish-core/lib/virtual_media.hpp | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/redfish-core/lib/virtual_media.hpp b/redfish-core/lib/virtual_media.hpp
index 717d5dae9b..b31856679b 100644
--- a/redfish-core/lib/virtual_media.hpp
+++ b/redfish-core/lib/virtual_media.hpp
@@ -670,6 +670,7 @@ inline void doMountVmLegacy(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
                             std::string&& userName, std::string&& password)
 {
     int fd = -1;
+    dbus::utility::DbusVariantType unixFd = -1;
     std::shared_ptr<CredentialsPipe> secretPipe;
     if (!userName.empty() || !password.empty())
     {
@@ -708,8 +709,10 @@ inline void doMountVmLegacy(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
     auto wrapper = doListenForCompletion(name, objectPath, action, true,
                                          asyncResp);
 
-    dbus::utility::DbusVariantType unixFd(
-        std::in_place_type<sdbusplus::message::unix_fd>, fd);
+    if(imageUrl.find("nfs://") != 0){
+    unixFd = dbus::utility::DbusVariantType(
+    std::in_place_type<sdbusplus::message::unix_fd>, fd);
+    }
 
     sdbusplus::message::object_path path(
         "/xyz/openbmc_project/VirtualMedia/Legacy");
-- 
2.34.1

